primitives/account/src/account/staking_contract/staker.rs

```diff
@@ -247,10 +312,10 @@ impl StakingContract {
             store.expect_validator(new_validator_address)?;
         }
 
-        // If stake is currently delegated, it must be inactive.
-        if let Some(validator_address) = &staker.delegation {
+        // If stake is currently delegated, than the active balance should be 0.
```

Comment:
Alternatively, `then`.
```suggestion
        // If stake is currently delegated, the active balance must be 0.
```